stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessagesLazyListState.kt

```diff
@@ -0,0 +1,157 @@
+
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.mapSaver
+import androidx.compose.ui.unit.IntSize
+import io.getstream.chat.android.client.models.Message
+import java.io.Serializable
+
+/**
+ * Provides a wrapper around lazy list state to be used with [Messages] composable. It is used to keep track of the
+ * focused message offset needed to center the focused message in the scroll list.
+ *
+ * @param lazyListState State of the lazy list that represents the list of messages. Useful for controlling the
+ * scroll state.
+ * @param messageOffsetHandler The handler that calculates the focused message offset.
+ */
+public data class MessagesLazyListState(
+    public val lazyListState: LazyListState,
+    private val messageOffsetHandler: MessageOffsetHandler = defaultOffsetHandler,
+) {
+
+    /**
+     * By default tracks the size of [Messages] composable. Used to calculate [focusedMessageOffset].
+     */
+    private val _parentSize: MutableState<IntSize> = mutableStateOf(IntSize.Zero)
+
+    /**
+     * Tracks the state of the focused [Message]. Used to calculate [focusedMessageOffset].
+     */
+    private val _focusedMessageSize: MutableState<IntSize> = mutableStateOf(IntSize.Zero)
+
+    /**
+     * The offset to be applied to a message after it has been marked for focus.
+     */
+    private val _focusedMessageOffset: MutableState<Int> = mutableStateOf(0)
+    public val focusedMessageOffset: Int by _focusedMessageOffset
+
+    /**
+     * Updates the list parent holder. By default will update the size of [Messages] composable.
+     *
+     * @param parentSize The size of the parent view.
+     */
+    public fun updateParentSize(parentSize: IntSize) {
+        if (parentSize == _parentSize.value) return
+        _parentSize.value = parentSize
+        calculateMessageOffset(parentSize, _focusedMessageSize.value)
+    }
+
+    /**
+     * Updates the size of the focused message.
+     *
+     * @param focusedMessageSize The size of the message view we wish to focus to on the screen.
+     */
+    public fun updateFocusedMessageSize(focusedMessageSize: IntSize) {
+        if (focusedMessageSize == _focusedMessageSize.value) return
+        _focusedMessageSize.value = focusedMessageSize
+        calculateMessageOffset(_parentSize.value, focusedMessageSize)
+    }
+
+    /**
+     * Calculates the offset of the focused message on the screen.
+     *
+     * @param parentSize The size of the parent view.
+     * @param focusedMessageSize The size of the message view we wish to focus to on the screen.
+     */
+    private fun calculateMessageOffset(parentSize: IntSize, focusedMessageSize: IntSize) {
+        _focusedMessageOffset.value = messageOffsetHandler.calculateOffset(parentSize, focusedMessageSize)
```

Comment:
Maybe `updateMessageOffset` would be the better naming scheme here in order to make the inner workings of the function more explicit and it seems to be a shorthand for doing so.

Reply:
I would leave it calculate since we are doing some calculations with the parameters we send, we are not just applying the passed value to the offset.

Reply:
https://github.com/GetStream/stream-chat-android/pull/3948#discussion_r998537998

We can but then it should be a constant we use for every check if the item is not insde the list. I would leave it as is.